### PR TITLE
Feature/535 show job template usage

### LINKED ISSUE #535: Show job template usage
UX - Show job template usage

### PR DESCRIPTION
**Pull request by jozefbakus**
Implements: #535 

Show job template usage

### CODE REVIEW

File: src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/services/JobTemplateService.scala [hunk 1]
```diff
@@ -35,6 +35,7 @@ trait JobTemplateService {
   def createJobTemplate(jobTemplate: JobTemplate)(implicit ec: ExecutionContext): Future[JobTemplate]
   def updateJobTemplate(jobTemplate: JobTemplate)(implicit ec: ExecutionContext): Future[JobTemplate]
   def deleteJobTemplate(id: Long)(implicit ec: ExecutionContext): Future[Boolean]
+  def getJobTemplateUsage(id: Long): Future[Seq[Workflow]]
```
Comment (kevinwallimann) [inline src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/services/JobTemplateService.scala]: I would name this `getWorkflowsByJobTemplate`, similar to `getWorkflowsByProjectName`. You can then ask yourself, whether this belongs to the `WorkflowService` or the `JobTemplateService`, but for me it's ok here.

File: ui/src/app/components/admin/job-templates/job-templates-home/job-template-usage/job-template-usage.component.html [hunk 2]
```diff
@@ -0,0 +1,38 @@
+<!--
```
Comment (kevinwallimann) [inline ui/src/app/components/admin/job-templates/job-templates-home/job-template-usage/job-template-usage.component.html:1]: This does look similar to the workflow grid. I feel a bit uneasy to "duplicate" this here and I'm wondering if it wouldn't be possible to reuse it? I mean, what are you going to do with that information? Wouldn't you want to deactivate, or export the listed workflows? Maybe we could add a workflows endpoint that takes the job template as a parameter, similar to the runs/{workflow-id} endpoint?

File: src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/services/JobTemplateService.scala (hunk 1 above)
Comment (jozefbakus) [reply src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/services/JobTemplateService.scala]: Renamed

File: ui/src/app/components/admin/job-templates/job-templates-home/job-template-usage/job-template-usage.component.html (hunk 2 above)
Comment (jozefbakus) [reply ui/src/app/components/admin/job-templates/job-templates-home/job-template-usage/job-template-usage.component.html:1]: I am not sure about this. What this shows: usage of the job template - where is used and if it is used actively.
With the current implementation, it is better to have it "duplicated" because some methods on table uses workflow actions.
I would rather go with a new workflows endpoint.

Comment (kevinwallimann) [reply ui/src/app/components/admin/job-templates/job-templates-home/job-template-usage/job-template-usage.component.html:1]: Maybe let me rephrase this. In principle, you could also add a column on the workflow table to find the usage of the job templates (It wouldn't be quite so easy because the job templates are on the dag definition). That's why for me, the job templates usage, is just a list of workflows. And I believe over time, we will want to do the same things on this list of workflows as on the main workflow grid.

Comment (jozefbakus) [reply ui/src/app/components/admin/job-templates/job-templates-home/job-template-usage/job-template-usage.component.html:1]: I would leave it on the templates page. 
The workflows main page requires some refactoring. First, we need to implement backend filtering and sorting and replace the left side list with only project list and after that we can enhance table with new column

**Review by kevinwallimann — APPROVED** [review]
